internal/commands/mountpoint_nfs.go

```diff
@@ -0,0 +1,208 @@
+//go:build !nonfs
+
+package commands
+
+import (
+	"encoding/json"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/djdv/go-filesystem-utils/internal/command"
+	"github.com/djdv/go-filesystem-utils/internal/filesystem"
+	p9fs "github.com/djdv/go-filesystem-utils/internal/filesystem/9p"
+	"github.com/djdv/go-filesystem-utils/internal/filesystem/nfs"
+	"github.com/djdv/go-filesystem-utils/internal/generic"
+	"github.com/multiformats/go-multiaddr"
+)
+
+type (
+	nfsHostSettings  nfs.Host
+	nfsHostOption    func(*nfsHostSettings) error
+	nfsHostOptions   []nfsHostOption
+	nfsGuestSettings struct {
+		nfs.Guest
+		defaultUID, defaultGID,
+		defaultHostname, defaultDirpath bool
+	}
+	nfsGuestOption  func(*nfsGuestSettings) error
+	nfsGuestOptions []nfsGuestOption
+)
+
+const nfsServerFlagName = "server"
+
+func (ns nfsGuestSettings) marshal(string) ([]byte, error) {
+	return json.Marshal(ns.Guest)
+}
+
+func makeNFSCommand() command.Command {
+	return makeMountSubcommand(
+		nfs.HostID,
+		makeGuestCommands[nfsHostOptions, nfsHostSettings](nfs.HostID),
+	)
+}
+
+func makeNFSHost(path ninePath, autoUnlink bool) (filesystem.Host, p9fs.MakeGuestFunc) {
+	guests := makeMountPointGuests[nfs.Host](path)
+	return nfs.HostID, newMakeGuestFunc(guests, path, autoUnlink)
+}
+
+func (on *nfsHostOptions) BindFlags(flagSet *flag.FlagSet) { /* NOOP */ }
+
+func (on nfsHostOptions) make() (nfsHostSettings, error) {
+	return makeWithOptions(on...)
+}
+
+func (*nfsHostOptions) usage(guest filesystem.ID) string {
+	return string(nfs.HostID) + " hosts " +
+		string(guest) + " as an NFS server"
+}
+
+func (set nfsHostSettings) marshal(arg string) ([]byte, error) {
+	if arg == "" {
+		err := command.UsageError{
+			Err: generic.ConstError(
+				"expected server multiaddr",
+			),
+		}
+		return nil, err
+	}
+	maddr, err := multiaddr.NewMultiaddr(arg)
+	if err != nil {
+		return nil, err
+	}
+	set.Maddr = maddr
+	return json.Marshal(set)
+}
+
+func unmarshalNFS() (filesystem.Host, decodeFunc) {
+	return nfs.HostID, func(b []byte) (string, error) {
+		var host nfs.Host
+		if err := json.Unmarshal(b, &host); err != nil {
+			return "", err
+		}
+		if maddr := host.Maddr; maddr != nil {
+			return host.Maddr.String(), nil
+		}
+		return "", errors.New("NFS host address was not present in the mountpoint data")
+	}
+}
+
+func makeNFSGuestCommand[
+	HC mountCmdHost[HT, HM],
+	HM marshaller,
+	HT any,
+](host filesystem.Host,
```

Comment:
I think this might be `gofumpt` rules being enabled in my LSP settings.
If I remove it, it just inserts it again.
I like most of the decisions of `gofumpt`, but it usually does weird stuff with function signatures...
I'll leave this alone for now and hope the entire function gets blown away in a future refactor.